src/main/java/com/google/javascript/clutz/ImportRenameMapBuilder.java

```diff
@@ -79,50 +88,71 @@ private String getGoogModuleId(Node astRoot) {
         String importedModuleId = statement.getFirstFirstChild().getChildAtIndex(1).getString();
         String variableName = statement.getFirstChild().getString();
-        String symbolIdInCurrentModule =
-            "module$contents$" + importingModuleId.replace(".", "$") + "_" + variableName;
-        String symbolIdInDefiningModule = "module$exports$" + importedModuleId.replace(".", "$");
-        importRenameMap.put(symbolIdInCurrentModule, symbolIdInDefiningModule);
+
+        String localSymbolName = buildLocalSymbolName(localModuleId, variableName);
+        String exportedSymbolName = buildExportedSymbolName(importedModuleId);
+        importRenameMap.put(localSymbolName, exportedSymbolName);
+        // `const {C, Clazz: RenamedClazz} = goog.require()`
       } else if (isGoogRequireDestructuringAssignment(statement)) {
         String importedModuleId =
             statement.getFirstChild().getChildAtIndex(1).getChildAtIndex(1).getString();
-        for (Node destructure : statement.getFirstFirstChild().children()) {
-          String originalName = destructure.getString();
+        for (Node destructured : statement.getFirstFirstChild().children()) {
+          String originalName = destructured.getString();
           // Destructuring can use the original name `const {A} = goog.require("foo.a")` or rename
-          // it `const {A: RenamedA} = ...`, and closure uses whichever in the symbole name it
+          // it `const {A: RenamedA} = ...`, and closure uses whichever in the symbol name it
           // generates, so we have to extract it.
           String variableName;
-          if (destructure.getFirstChild() != null) {
-            variableName = destructure.getFirstChild().getString();
+          if (destructured.getFirstChild() != null) {
+            variableName = destructured.getFirstChild().getString();
           } else {
             variableName = originalName;
           }
-          String symbolIdInCurrentModule =
-              "module$contents$" + importingModuleId.replace(".", "$") + "_" + variableName;
-          String symbolIdInDefiningModule =
-              "module$exports$" + importedModuleId.replace(".", "$") + "_" + originalName;
-          importRenameMap.put(symbolIdInCurrentModule, symbolIdInDefiningModule);
+
+          String localSymbolName = buildLocalSymbolName(localModuleId, variableName);
+          String exportedSymbolName = buildExportedSymbolName(importedModuleId, originalName);
+          importRenameMap.put(localSymbolName, exportedSymbolName);
         }
       }
     }
 
     return importRenameMap;
   }
+
+  /**
+   * The exported symbol can take 2 forms - one where it refers to everything that the module
+   * exports and another where it refers to just one thing the module exports. If there's a normal
+   * import, the symbol name is just the module name.
+   */
+  private String buildExportedSymbolName(String importedModuleId) {
```

Comment:
maybe use a different method name that reflects the situation it is used for, instead of overloading based on parameters?

Reply:
Done.